packages/shared/lib/clients/provider.client.ts

```diff
@@ -58,18 +63,30 @@ class ProviderClient {
 
         const credentials = connection.credentials as OAuth2Credentials;
 
-        if (!credentials.refresh_token) {
+        if (!credentials.refresh_token && !credentials.access_token) {
```

Comment:
what if we are trying to refresh Braintree token and there is no credentials.refresh_token but an access_token. Is that a possible scenario. If yes it would not enter the `if` even though it should

Reply:
You're right. To address this, I will update the logic to explicitly check for both scenarios.